File: src/p1304/cpp/solution.cpp
class Solution {
public:
    vector<int> sumZero(int n) {
        vector<int> result;
        if (n % 2 != 0) {
            result.push_back(0);
            n--;
        }
        while (n > 0) {
            result.push_back(n);
            result.push_back(-n);
            n -= 2;
        }
        return result;
    }
};

File: src/p1304/python/solution.py
class Solution:
    def sumZero(self, n: int) -> List[int]:
        result = [0] * n
        if n % 2 != 0:
            n -= 1
            result[n] = 0
        while n > 0:
            result[n - 1] = n
            result[n - 2] = -n
            n -= 2
        return result
